### ISSUE TITLE
Adding a missing chuck script to the VM confuses subsequent error messages

### ISSUE THREAD

**Issue by ynohtna** [ccrma/chuck#188]
Small test case like this:
```
Machine.add("./missing/script-that-does-not-exist");
MidiIn min;
min.open(31337);
```
results in the following output (both with Chuck CLI and MiniAudicle), where Chuck incorrectly attributes the MIDI open error to the `script-that-does-not-exist` file:
```
[script-that-does-not-exist]: no such file or directory
[script-that-does-not-exist]: MidiIn: couldn't open MIDI port 31337...
```

This is pretty low priority, as adding missing or typo-ed script filenames to the VM is obviously developer error, but the consequence on subsequent error messages can definitely lead to some befuddled head scratching.

**Comment by gewang** [comment]
I don't see the issue here?  the "no such file or directory" error is from the Machine.add() and the MidiIn error from opening a bogus MIDI device?

**Comment by ynohtna** [comment]
Sorry for the confusion, I will try to be clearer: the (expected) MIDI error is reported as if it is part of the script that does not exist (and failed to be added to the VM), rather than the actual name of the file in which the error was triggered.

Commenting out the Machine.add line changes the MIDI error message's leading `[script-that-does-not-exist]` to `[Chuck`:
```
[chuck]: MidiIn couldn't open MIDI port 31337..
```
Which still does not attribute the error correctly to its host script filename, but does indicate that the error occurred within the "root" script.

**Comment by ynohtna** [comment]
The TODO comment here about g_program not being reset in the cleanup branch seems a strong clue:
https://github.com/ccrma/chuck/blob/091abe0d1490c76a28020b72c19319838a8a4a2d/src/core/chuck_parse.cpp#L200-L219

I'll tinker and see if I can naively fix this.

**Comment by ynohtna** [comment]
Actually, it seems more to do with the preceding lines that strcpy the filename in g_filename not being reset in the cleanup:
https://github.com/ccrma/chuck/blob/091abe0d1490c76a28020b72c19319838a8a4a2d/src/core/chuck_parse.cpp#L181-L189

**Comment by gewang** [comment]
Ah I see -- thanks for the tracking this down.  This has been addressed in 050a1461fdc5fcb478ab6bc5bd72b7f1fca7f8b1 — and being smuggled into the 1.4.1.0 release (happening today).  Thank you!
